doc/help/elektra-data-structures.md

```diff
@@ -366,3 +366,80 @@ The algorithm uses the `syncbits` to decide if the key set needs to be
 synchronised.
 
 Continue reading [with the error handling](elektra-error-handling.md).
+
+## Order Preserving Minimal Perfect Hash Map
+
+### Vstack
+
+The `Vstack` is a virtual stack implementation with dynamical memory allocation.
+The allocation doubles the space if the `Vstack` is full and reduces the space by the half
+if the memory usage gets under a quarter.
+
+The data will be stored as void pointer and can be inserted with the push call.
+Retrieve is possible through the pop call.
+Besides the classical init and del functions, an is empty check is also implemented.
+
+The typedef `Vstack` contains the following fields:
+
+	typedef struct
+	{
+		size_t minSize;
+		size_t size;
+		void ** data;
+		void ** head;
+	} Vstack;
+
+- **minSize**: Is the minimum size of the `Vstack`. After each pop a check is performed to see if a shrink is needed.
+This field defines the minimum size, which the `Vstack` size will never get below.
+
+- **size**: Is the actual allocated size of the `Vstack`
+
+- **data**: Is the array in which the data will be stored.
+
+- **head**: Is the stack pointer for fast insertion and retrieve. After each push or pop the stack pointer gets incremented or decremented.
```

Comment:
Again: maybe this additional docu is better placed in the doxygen comments directly at the struct.